controllers/ProfileController.js

```diff
@@ -0,0 +1,94 @@
+import models from '../database/models';
+import ServerResponse from '../modules';
+
+const { successResponse, errorResponse } = ServerResponse;
+/**
+ * @class ProfileController
+ * @exports ProfileController
+ */
+export default class ProfileController {
+  /**
+   *
+   * @method createProfile
+   * @description Create a profile for registered user
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {object} returns profile info
+   * @memberof ProfileController
+   */
+  static async createProfile(req, res) {
+    const { avatar, bio } = req.body;
+    const { id } = req.user;
+    const profileExist = await models.Profile.findOne({ where: { userId: id } });
+    if (profileExist) {
+      return errorResponse(res, 409, 'Profile already exist');
+    }
+    await models.Profile.create({
+      userId: id,
+      avatar,
+      bio,
+    });
+    const { firstName, lastName } = req.user;
+    const resData = {
+      message: 'Your profile has been created successfully',
+      profile: {
+        name: `${firstName} ${lastName}`,
+        bio,
+        avatar,
+      },
+    };
+    return successResponse(res, 201, resData);
+  }
+
+  /**
+   *
+   * @method editProfile
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {object} returns profile info
+   */
+  static async editProfile(req, res) {
+    const {
+      avatar, bio, firstName, lastName,
+    } = req.body;
+    const { id } = req.user;
+    await models.Profile.update(
+      { bio, avatar },
+      { where: { userId: id } },
+    );
+
+    await models.User.update(
+      { firstName, lastName },
+      { where: { id } },
+    );
+    const userData = await models.User.findOne({ where: { id } });
+    const resData = {
+      message: 'Your profile has been updated successfully',
+      profile: {
+        avatar,
+        bio,
+        firstName: userData.dataValues.firstName,
+        lastName: userData.dataValues.lastName,
+      },
+    };
+    return successResponse(res, 200, resData);
+  }
+
+  /**
+   *
+   * @method viewProfile
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {object} returns profile info
+   */
+  static async viewProfile(req, res) {
+    const { id } = req.params;
+    const profileData = await models.User.findOne({ where: { id } });
```

Comment:
1. Is a user's profile saved n the `User` model?
2. What happens if there is no profile with a given `id`?

Reply:
I have a validation to check if the id exist in the database.

Reply:
Answer to 1. I would edit that., Thanks